app/controllers/codelijsten/edit.js

```diff
@@ -361,137 +388,116 @@ export default class CodelijstenEditController extends Controller {
+      this.dbConceptScheme.label == this.schemeName &&
       !this.isConceptListChanged() &&
       this.conceptsToDelete.length == 0
     );
   }
 
   isConceptListChanged() {
-    return isConceptArrayChanged(this.conceptsInDatabase, this.concepts);
+    return isConceptArrayChanged(this.dbConcepts, this.conceptList);
   }
 
-  mapConceptModels(concepts) {
-    return concepts.map((concept) => {
-      return {
-        id: concept.id,
-        label: concept.label,
-      };
-    });
-  }
-
-  async getConceptSchemeById(conceptSchemeId) {
-    try {
-      const conceptScheme = await this.store.findRecord(
-        'concept-scheme',
-        conceptSchemeId,
-        {
-          include: 'concepts',
-        }
-      );
-      await conceptScheme.reload();
+  get isReadOnly() {
+    if (!this.dbConceptScheme) return true;
 
-      return conceptScheme;
-    } catch (error) {
-      throw this.intl.t('constraints.couldNotGetConceptSchemeWithId', {
-        id: conceptSchemeId,
-      });
-    }
+    return this.isPrivateConceptScheme || this.isArchivedConceptScheme;
   }
 
-  @action
-  opposite(boolean) {
-    if (typeof boolean !== 'boolean') {
-      showErrorToasterMessage(
-        this.toaster,
-        this.intl.t('messages.error.valueMustBeBoolean', {
-          value: boolean,
-        })
-      );
-      return false;
-    }
+  get isPrivateConceptScheme() {
+    if (!this.dbConceptScheme) return true;
 
-    return !boolean;
+    return !this.dbConceptScheme.isPublic;
   }
 
-  @action
-  async discardSave() {
-    this.codelistName = this.conceptScheme.label;
-    this.codelistDescription = this.conceptScheme.description;
-    this.concepts = await this.conceptScheme.getConceptModels();
-    this.conceptsToDelete = [];
-    this.nameErrorMessage = null;
-    this.descriptionErrorMessage = null;
+  get isArchivedConceptScheme() {
+    if (!this.dbConceptScheme) return false;
 
-    this.isSaveModalOpen = false;
-    this.setIsSaveButtonDisabled();
+    return this.dbConceptScheme.isArchived;
+  }
 
-    this.goToNextRoute();
+  get canArchiveCodelist() {
+    return (
+      !this.isPrivateConceptScheme &&
+      this.hasConcepts &&
+      this.isValidConceptSchemeName() &&
+      !this.isArchivedConceptScheme &&
+      this.isSaveDisabled
+    );
   }
 
-  @action
-  goToNextRoute() {
-    this.router.transitionTo(this.nextRoute);
+  get canExportCodelist() {
+    return (
+      !this.isPrivateConceptScheme &&
+      this.hasConcepts &&
+      this.isValidConceptSchemeName() &&
+      this.isSaveDisabled
+    );
   }
 
-  showSaveModal(nextRoute) {
-    this.isSaveModalOpen = true;
-    this.nextRoute = nextRoute;
+  get splitButtonHasActiveActions() {
+    return this.canArchiveCodelist || this.canExportCodelist;
   }
 
-  resetErrors() {
-    this.nameErrorMessage = null;
-    this.descriptionErrorMessage = null;
-    this.isDuplicateName = false;
+  get hasConcepts() {
```

Comment:
rename to _hasConceptsInList_
Think this would be more clear that it points to the _conceptList_ variable where all the concepts are in that are shown in the UI